src/System.Private.Interop/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -787,19 +825,32 @@ public static unsafe IntPtr StringToCoTaskMemUni(String s)
                     throw new ArgumentOutOfRangeException(nameof(s));
 
                 IntPtr hglobal = PInvokeMarshal.CoTaskMemAlloc(new UIntPtr((uint)nb));
-
-                if (hglobal == IntPtr.Zero)
+                fixed (char* firstChar = s)
                 {
-                    throw new OutOfMemoryException();
```

Comment:
Why do we no longer need this?

Reply:
PInvokeMarshal.CoTaskMemAlloc has this check & throw already.

Reply:
Actually, it has it on Linux only. It does not have it on Windows.